### PR TITLE
[MAINT] Switch to setup.cfg-based configuration

### LINKED ISSUE #872: [MAINT] Change setup configuration
### Summary
The setup for python packages has changed over the last few years. There is now the option of a `pyproject.toml` (which tools like `poetry` use), a `setup.cfg` (which is what pip is designed to use), and there is now a circular dependency from `info.py` if we leave `duecredit` there but wish to remove `due.py`. We should refine the setup configuration.

### Next Steps
- [ ] Pick which items belong in `setup.cfg`
- [ ] Potentially: switch to pyproject.toml

### PR DESCRIPTION
**Pull request by tsalo**
Closes #872.

Changes proposed in this pull request:

- Delete `tedana/info.py` and move its contents into `setup.cfg`.

### CODE REVIEW

File: tedana/__init__.py [hunk 1]
```diff
@@ -54,3 +45,5 @@
     path="tedana",
     cite_module=True,
 )
+
+del get_versions
```
Comment (jbteves) [inline tedana/__init__.py:49]: Why is this needed?

Comment (tsalo) [reply tedana/__init__.py:49]: It's not needed, but I think it's nice to have a clean set of locals for the package. I can remove it though.

Comment (jbteves) [reply tedana/__init__.py:49]: I see. We can leave it, just wanted to make sure I understood why it was there.

**Review by jbteves — APPROVED** [review]

**Review by eurunuela — APPROVED** [review]
LGTM!
